scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -54,8 +67,62 @@ void analyze() {
     @Override
     public void visit(CheckExpressionNode checkExpressionNode) {
         if (checkExpressionNode.checkKeyword().kind().equals(SyntaxKind.CHECKPANIC_KEYWORD)) {
-            scannerContext.getReporter().reportIssue(document, checkExpressionNode.location(),
-                    CoreRule.AVOID_CHECKPANIC.rule());
+            reportIssue(scannerContext, document, checkExpressionNode, CoreRule.AVOID_CHECKPANIC.rule());
+        }
+    }
+
+    @Override
+    public void visit(FunctionSignatureNode functionSignatureNode) {
+        functionSignatureNode.parameters().forEach(parameter -> {
+            if (parameter instanceof IncludedRecordParameterNode includedRecordParameterNode) {
+                includedRecordParameterNode.paramName().ifPresent(name -> {
+                    if (isUnusedNode(name, semanticModel)) {
+                        reportIssue(scannerContext, document, name, CoreRule.UNUSED_FUNCTION_PARAMETERS.rule());
+                    }
+                });
+            } else {
+                if (isUnusedNode(parameter, semanticModel)) {
+                    reportIssue(scannerContext, document, parameter, CoreRule.UNUSED_FUNCTION_PARAMETERS.rule());
+                }
+            }
+            this.visitSyntaxNode(parameter);
+        });
+        functionSignatureNode.returnTypeDesc().ifPresent(returnTypeDesc -> returnTypeDesc.accept(this));
+    }
+
+    @Override
+    public void visit(ImplicitAnonymousFunctionExpressionNode implicitAnonymousFunctionExpressionNode) {
+        checkUnusedParametersInImplicitFunctionExpression(implicitAnonymousFunctionExpressionNode.params());
+        this.visitSyntaxNode(implicitAnonymousFunctionExpressionNode.expression());
+    }
+
+    private void checkUnusedParametersInImplicitFunctionExpression(Node params) {
+        if (params instanceof ImplicitAnonymousFunctionParameters parameters) {
+            parameters.parameters().forEach(parameter -> {
+                if (isUnusedNode(parameter, semanticModel)) {
+                    reportIssue(scannerContext, document, parameter, CoreRule.UNUSED_FUNCTION_PARAMETERS.rule());
+                }
+            });
+            return;
+        }
+
+        if (params instanceof SimpleNameReferenceNode) {
+            if (isUnusedNode(params, semanticModel)) {
+                reportIssue(scannerContext, document, params, CoreRule.UNUSED_FUNCTION_PARAMETERS.rule());
+            }
         }
     }
+
+    private void reportIssue(ScannerContext scannerContext, Document document, Node node, Rule rule) {
+        scannerContext.getReporter().reportIssue(document, node.location(), rule);
+    }
+
+    private boolean isUnusedNode(Node node, SemanticModel semanticModel) {
```

Comment:
Do we need to pass `semanticModel` as an agrument here? As far as I can see it always the final field